pillar/vault.py

```diff
@@ -169,6 +169,29 @@ def _authenticate(conn):
         conn.token = os.environ.get('VAULT_TOKEN')
 
 
+def couple(location, conn):
+    coupled_data = {}
+    if isinstance(location, basestring):
+        try:
+            (path, key) = location.split('?', 1)
+        except ValueError:
+            (path, key) = (location, None)
+        secret = conn.read(path)
+        if key:
+            secret = secret["data"].get(key, None)
+            prefix = "base64:"
+            if secret.startswith(prefix):
+                secret = base64.b64decode(secret[len(prefix):]).rstrip()
+        if secret or not CONF["unset_if_missing"]:
```

Comment:
I guess it's the top two cases that get me... how are we unsetting the key in the resulting dict?  It seems like this option should be checked in the caller -- which then can actually opt to not create the entry in the dictionary -- otherwise it'll be there (as None or '') irrespective of the unset_if_missing option.